src/main/java/org/broadinstitute/hellbender/tools/walkers/haplotypecaller/HaplotypeCaller.java

```diff
@@ -202,9 +202,7 @@ public void onTraversalStart() {
     }
 
     private static CachingIndexedFastaSequenceFile getReferenceReader(ReferenceInputArgumentCollection referenceArguments) {
-        // TODO: this code is duplicated in AssemblyBasedCallerUtils
-        final Path reference = IOUtils.getPath(referenceArguments.getReferenceFileName());
-        return new CachingIndexedFastaSequenceFile(reference);
+        return new CachingIndexedFastaSequenceFile(referenceArguments.getReferencePath());
```

Comment:
Is there a reason why the `CachingIndexedFastaSequenceFile` can't accept a `GATKPathSpecifier` directly? (since it lives in GATK, even though its parent class is in HTSJDK)

Reply:
Sounds right - I'll add a constructor.

Reply:
Actually, I started going down this path and it would enable ReferenceFileSource to use GATKPathSpecifier as well. I'm going to do the simple change and add the constructor for this one case in this PR, and then do the rest in another PR.